### PR TITLE
audio-device: update system effects when output changes

### LINKED ISSUE #4706: [Set Audio Device] Changing audio device does not affect sound effects
### Extension

https://www.raycast.com/benvp/audio-device

### Description

Changing the audio device through the extension changes the "Play sound effects through" setting from "Selected Sound Output Device" (sound effects play through the active audio device) to whatever the previous audio output was before switching. This results in sound effects always playing through the first device that was in use before using the extension until the setting is changed back to "Selected Sound Output Device" in System Settings.

### Steps To Reproduce

1. Switch audio device using the extension
2. Observe that the "Play sound effects through" setting in System Settings has changed from "Selected Sound Output Device" to the previous audio device in use before switching

### Current Behaviour

"Play sound effects through" setting is affected by the extension

### Expected Behaviour

"Play sound effects through" setting remains untouched

### PR DESCRIPTION
**Pull request by rcruzper**
## Description

Add an option to update "Play sound effects through" in Sound Preferences when output device changes

## Screencast

Screenshot 2023-05-10 at 18 56 21

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

resolves #4706

### CODE REVIEW

File: extensions/audio-device/src/helpers.tsx [hunk 1]
```diff
@@ -130,6 +131,11 @@ function SetAudioDeviceAction({ device, type }: SetAudioDeviceActionProps) {
   );
 }
 
+function setOutputAndSystemDevice(deviceId: string) {
+  setDefaultOutputDevice(deviceId);
+  setDefaultSystemDevice(deviceId);
+}
```
Comment (mathieudutour) [inline extensions/audio-device/src/helpers.tsx:137]:
```suggestion
async function setOutputAndSystemDevice(deviceId: string) {
  await setDefaultOutputDevice(deviceId);
  await setDefaultSystemDevice(deviceId);
}
```

**Comment by FezVrasta** [comment]
Should this be a user preference?

**Comment by rcruzper** [comment]
> Should this be a user preference?

in my use case the system sounds should always be listened from the current output device. Which use cases do you think would need that user preference?

**Comment by FezVrasta** [comment]
I switch the audio to the headphones during calls but I don't want the system sounds there. Same when I listen to music on my HomePod.

**Comment by rcruzper** [comment]
Makes sense, I will add the preference.

On the other hand, the current audio-device binary asset used to change output and input device is only compatible with intel chips and I have a M2 chip so every time I want to make tests or used the extension I need to compile as universal binary, I want to change the package.json to use another library which compiles with an universal binary [here](https://github.com/spotxyz/macos-audio-devices/blob/master/package.json#L25). Do you think I should create another PR o can I change it here.

**Comment by rcruzper** [comment]
updated with the user preference, do you think is there any better naming for title and description?

**Review by FezVrasta — APPROVED** [review]

**Review by mathieudutour — APPROVED** [review]
Looks good, thanks!

**Pull request merged**